input.go

```diff
@@ -59,20 +115,24 @@ func InputsUTXORefsUniqueValidator() InputsValidatorFunc {
 	}
 }
 
-// InputsUTXORefIndexBoundsValidator returns a validator which checks that the UTXO ref index is within bounds.
-func InputsUTXORefIndexBoundsValidator() InputsValidatorFunc {
-	return func(index int, input *UTXOInput) error {
-		if input.TransactionOutputIndex < RefUTXOIndexMin || input.TransactionOutputIndex > RefUTXOIndexMax {
+// InputsSyntacticalIndicesWithinBounds returns an InputsSyntacticalValidationFunc which checks that the UTXO ref index is within bounds.
+func InputsSyntacticalIndicesWithinBounds() InputsSyntacticalValidationFunc {
+	return func(index int, input Input) error {
+		ref, is := input.(IndexedUTXOReferencer)
+		if !is {
```

Comment:
Shouldn't we return an error here?
If that line is valid, please add a comment why this is fine.

Reply:
9b091b3f600ae9f24643c269f90c3f2965a40f23